File: 3090-maximum-length-substring-with-two-occurrences/3090-maximum-length-substring-with-two-occurrences.java
class Solution {
    public int maximumLengthSubstring(String s) {

        int[] count = new int[26];
        int left = 0;
        int maxLength = 0;

        for (int right = 0; right < s.length(); right++) {
            int index = s.charAt(right) - 'a';
            count[index]++;

            while (count[index] > 2) {
                int leftIndex = s.charAt(left) - 'a';
                count[leftIndex]--;
                left++;
            }
            maxLength = Math.max(maxLength, right - left + 1);
        }
        return maxLength;
    }
}

File: 3090-maximum-length-substring-with-two-occurrences/3090-maximum-length-substring-with-two-occurrences.py
class Solution:
    def maximumLengthSubstring(self, s: str) -> int:
        count = {}
        left = 0
        max_length = 0

        for right in range(len(s)):
            count[s[right]] = count.get(s[right], 0) + 1

            while count[s[right]] > 2:
                count[s[left]] -= 1
                left += 1
                
            max_length = max(max_length, right - left + 1)

        return max_length